src/syntax/Scanner.js

```diff
@@ -275,9 +275,12 @@ export class Scanner {
   /**
    * @return {Token}
    */
-  peekToken(opt_index) {
-    // Too hot for default parameters.
-    return opt_index ? peekTokenLookahead() : peekToken();
+  peekToken() {
+    return peekToken();
+  }
+
+  peekTokenLookahead() {
+    return peekTokenLookahead();
```

Comment:
Seems like this would be better as a module exporting functions than a class-that's-not-really-a-class.  Then instead of the Parser calling `this.scanner_.peekToken()`, just `peekToken()`.